PYTHON-5768: AGENTS.md standardization for all DBX Repositories (#2819)
mongodb/mongo-python-driver: .github/AGENTS.md

# AGENTS.md - PyMongo (mongo-python-driver)

## Overview
Official Python driver for MongoDB.

## Tech Stack
- **Build/env:** `hatch` — use `hatch run test:test` and `hatch run test:test-async`, not `pytest` directly
- **Task runner:** `just` (Justfile at repo root)
- **Linter/formatter:** `ruff` via pre-commit
- **Type checker:** `mypy` (strict)
- **Docs:** Sphinx / reStructuredText; use Sphinx docstring format for all docstrings

> **Never** run `pip install bson`. PyMongo ships its own `bson`; the PyPI `bson` package silently breaks things.

## Project Structure

> - `pymongo/` — sync driver (MongoClient, Collection, Database, etc.)

Reviewer: ```suggestion
- `pymongo/synchronous/` — sync driver (MongoClient, Collection, Database, etc.); generated from `pymongo/asynchronous/` using unasync
```

> - `pymongo/asynchronous/` — async API, mirrors `pymongo/`

Reviewer: ```suggestion
- `pymongo/asynchronous/` — async API, mirrors `pymongo/synchronous/`
```

- `bson/` — BSON implementation
- `gridfs/` — GridFS API
- `test/` — sync test suite
- `test/asynchronous/` — async test suite, mirrors `test/`
- `test/unified_format/` — cross-driver spec tests (Unified Test Format)
- `tools/convert_test_to_async.py` — generates async test stubs from sync tests

## Commands
```bash
just install          # set up venv + pre-commit hook
just typing-mypy      # type check
just run lint-manual  # ruff linter

hatch run test:test         # run sync tests
hatch run test:test-async   # run async tests

MONGODB_VERSION=8.0 just run-server  # start a local MongoDB server
```

## Testing
Tests require a live MongoDB server (`just run-server` above).

> When modifying sync code, also update the async counterpart:

Reviewer: ```suggestion
Never modify pymongo/synchronous code first; update the async counterpart and run our `tools/synchro.py` script:
```

> 1. Make changes in `pymongo/` and `test/`.

Reviewer: ```suggestion
1. Make changes in `pymongo/asynchronous` or top-level `pymongo/` files and `test/`.
```
